src/GUI/EFCorePowerTools/Handlers/ReverseEngineer/EfRevEngLauncher.cs

```diff
@@ -123,13 +125,26 @@ private bool IsDotnetInstalled()
             var startInfo = new ProcessStartInfo
             {
                 FileName = "dotnet",
-                Arguments = "--version",
+                Arguments = "--list-sdks",
             };
 
             var result = RunProcess(startInfo);
 
-            return result.StartsWith("3.", StringComparison.OrdinalIgnoreCase)
-                || result.StartsWith("5.", StringComparison.OrdinalIgnoreCase);
+            var isInstalled = false;
+            var sdks = result.Split(new string[] { "\r\n" }, StringSplitOptions.RemoveEmptyEntries);
+            foreach (var item in sdks)
+            {
+                isInstalled = useEFCore5 ?
```

Comment:
The tool just needs either dotnet 3 or dotnet 5, useEFCore5 is not relevant here.

Reply:
If the user uses EFCore3.0, the 3.0 SDK needs to be installed on the machine. If the user uses EFCore5.0, the 5.0 SDK needs to be installed. It will not work without the corresponding SDK

Reply:
EF Core 5 works fine with .NET 3.1! and this is for running the reverse engineer tool, not for runtime concerns

Reply:
https://github.com/ErikEJ/EFCorePowerTools/blob/master/src/GUI/efpt30.core/efpt30.core.csproj#L5

```
.NET SDK (反映任何 global.json):
 Version:   5.0.101
 Commit:    d05174dc5a

运行时环境:
 OS Name:     Windows
 OS Version:  10.0.19042
 OS Platform: Windows
 RID:         win10-x86
 Base Path:   C:\Program Files (x86)\dotnet\sdk\5.0.101\

Host (useful for support):
  Version: 6.0.0-alpha.1.20560.10
  Commit:  72b7d236ad

.NET SDKs installed:
  5.0.101 [C:\Program Files (x86)\dotnet\sdk]
  6.0.100-alpha.1.20562.2 [C:\Program Files (x86)\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 5.0.1 [C:\Program Files (x86)\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.0-alpha.1.20526.6 [C:\Program Files (x86)\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 5.0.1 [C:\Program Files (x86)\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.0-alpha.1.20560.10 [C:\Program Files (x86)\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 5.0.1 [C:\Program Files (x86)\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.0-alpha.1.20560.7 [C:\Program Files (x86)\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download
```

![image](https://user-images.githubusercontent.com/5714438/103614851-7bf66000-4f64-11eb-8438-bfb3a8523bac.png)

![image](https://user-images.githubusercontent.com/5714438/103614899-929cb700-4f64-11eb-865c-aee4c9e784ff.png)

Reply:
https://github.com/ErikEJ/EFCorePowerTools/blob/master/src/GUI/efpt30.core/efpt30.core.csproj#L5 has nothing to do with reverse engineering

Reply:
efreveng and efrevneg50 are the reverse engineering tools

Reply:
Just remove the useEFCore5 check

Reply:
If useEFCore5 is checked, whether to use efreveng50 reverse engineering, if so, there is no 5.0 runtime on the machine (currently it is unlikely, because VS already includes 5.0 SDK), how does efreveng50 work normally?

Reply:
efreveng50 is a .NET Core 3.1 app (as you can see in the .csproj)

Reply:
Sorry, I missed it. Is it enough to check whether 3.1 is installed?

Reply:
3, 5 or 6 will work (as in the previous implementation) - newer runtimes can run older binaries.

Reply:
You can also change the check to "3.1" if you want, yes

Reply:
As shown in the figure above, there is no SDK for 3.1 on the machine, and the tool cannot work normally

Reply:
Are you testing with your modified code?

Reply:
Looking at the error message, that does not look like it is the case.

Reply:
Not my modified version, but the version downloaded from the market for testing

Reply:
Well, add the check for 5.0 and show me the result of testing your local changes then.